### PR TITLE
Check array/number bounds

### PR DESCRIPTION
**Pull request by ganymedes01**

### CODE REVIEW

File: src/make.ts [hunk 1]
```diff
@@ -181,23 +181,19 @@ export function makeString(
...
 }
 
-function checkNumber(value: any): Make<number> {
-  if (typeof value !== 'number') {
-    return getErrorWithValueMsg('expected a number', value);
-  }
-  return Make.ok(value);
-}
-
-export function makeNumber(value?: number): Maker<number, number> {
```
Comment (ganymedes01) [inline src/make.ts:191]: I checked and oats never generates `makeNumber` by giving it a parameter, so we get simplify this code

File: src/make.ts [hunk 2]
```diff
@@ -244,11 +240,17 @@ export function makeBoolean() {
   return checkBoolean;
 }
 
-export function makeArray(maker: any) {
+export function makeArray(maker: any, minSize?: number, maxSize?: number) {
   return (value: any, opts?: MakeOptions) => {
     if (!Array.isArray(value)) {
       return getErrorWithValueMsg('expected an array', value);
     }
+    if (minSize != null && value.length < minSize) {
```
Comment (ganymedes01) [inline src/make.ts:248]: openapi doesn't specify if this value is inclusive or exclusive 🤷‍♂️

Comment (sakari) [reply src/make.ts:248]: they say its same semantics as json-schema which seems to have it inclusive https://json-schema.org/understanding-json-schema/reference/array.html#length

File: src/make.ts (hunk 1 above)
Comment (sakari) [reply src/make.ts:191]: note even for enums? good catch

**Review by sakari — APPROVED** [review]

**Pull request merged**